test/config_tests.ml

```diff
@@ -0,0 +1,124 @@
+let pmsg =
+  Alcotest.testable (fun ppf (`Msg s) -> Fmt.pf ppf "Error @[<v>(%s)@]" s)
+    (=)
+
+let result_p_err (type x) ok =
+  (* like Alcotest.result, but pretty-prints the Error `Msg and
+     fails the test when [Error _] is encountered.
+     On success, [ok]'s [pp] and [equal]
+     function are used to pretty-print and determine test result. *)
```

Comment:
hmm, why not use `result <ok> p_msg` (with `p_msg` defined above) at call-sites?

Reply:
I had some kind of horrible hack going on locally where I kept track of which one was expected/actual and colored them differently because I found it quite hard to visually tell the difference between two long config files, but I think you're right.